pkg/agent/util/net_windows.go

```diff
@@ -1097,3 +1204,90 @@ func renameHostInterface(oriName string, newName string) error {
 	_, err := runCommand(cmd)
 	return err
 }
+
+func parseAdapters(name string) ([]Adapter, error) {
+	aas, err := adapterAddresses()
+	if err != nil {
+		return nil, err
+	}
+	var adapters []Adapter
+	for _, aa := range aas {
+		ifName := windows.UTF16PtrToString(aa.FriendlyName)
+		if ifName != name {
+			continue
+		}
+		index := aa.IfIndex
+		if index == 0 { // ipv6IfIndex is a substitute for ifIndex
+			index = aa.Ipv6IfIndex
+		}
+		ifi := net.Interface{
+			Index: int(index),
+			Name:  ifName,
+		}
+		if aa.OperStatus == windows.IfOperStatusUp {
+			ifi.Flags |= net.FlagUp
+		}
+		// For now we need to infer link-layer service
+		// capabilities from media types.
+		// TODO: use MIB_IF_ROW2.AccessType now that we no longer support
+		// Windows XP.
+		switch aa.IfType {
+		case windows.IF_TYPE_ETHERNET_CSMACD, windows.IF_TYPE_ISO88025_TOKENRING, windows.IF_TYPE_IEEE80211, windows.IF_TYPE_IEEE1394:
+			ifi.Flags |= net.FlagBroadcast | net.FlagMulticast
+		case windows.IF_TYPE_PPP, windows.IF_TYPE_TUNNEL:
+			ifi.Flags |= net.FlagPointToPoint | net.FlagMulticast
+		case windows.IF_TYPE_SOFTWARE_LOOPBACK:
+			ifi.Flags |= net.FlagLoopback | net.FlagMulticast
+		case windows.IF_TYPE_ATM:
+			ifi.Flags |= net.FlagBroadcast | net.FlagPointToPoint | net.FlagMulticast // assume all services available; LANE, point-to-point and point-to-multipoint
+		}
+		if aa.Mtu == 0xffffffff {
+			ifi.MTU = -1
+		} else {
+			ifi.MTU = int(aa.Mtu)
+		}
+		if aa.PhysicalAddressLength > 0 {
+			ifi.HardwareAddr = make(net.HardwareAddr, aa.PhysicalAddressLength)
+			copy(ifi.HardwareAddr, aa.PhysicalAddress[:])
+		}
+		adapter := Adapter{
+			Interface:     ifi,
+			CompartmentID: aa.CompartmentId,
+		}
+		adapters = append(adapters, adapter)
+	}
+	return adapters, nil
+}
+
+// adapterAddresses returns a list of IP adapter and address
+// structures. The structure contains an IP adapter and flattened
+// multiple IP addresses including unicast, anycast and multicast
+// addresses.
+// This function is copied from go/src/net/interface_windows.go, the difference is flag
+// GAA_FLAG_INCLUDE_ALL_COMPARTMENTS is introduced to query interfaces in all compartments.
```

Comment:
Consider moving `GAA_FLAG_INCLUDE_ALL_COMPARTMENTS` to be just before this func (as you put some types before their funcs).